src/Factors/GenericFunctions.jl

```diff
@@ -55,12 +55,16 @@ end
 
 DFG.getManifold(f::ManifoldFactor) = f.M
 
-function getSample(cf::CalcFactor{<:ManifoldFactor}, N::Int=1)
+function getSample(cf::CalcFactor{<:ManifoldFactor{M,Z}}) where {M,Z}
     #TODO @assert dim == cf.factor.Z's dimension
     #TODO investigate use of SVector if small dims
-    ret = [rand(cf.factor.Z) for _ in 1:N]
+    if M isa ManifoldKernelDensity
```

Comment:
suggest doing this with dispatch while figuring out `sample`, `rand`, `getSample`, `sampleTangent`

Reply:
It's a bit of an odd one out, so I quickly did the if to get it to run. I still want to try and avoid 2 `getSample` functions on this one. still TODO

Reply:
Sounds good